addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/handler/FreeboxHandler.java

```diff
@@ -403,272 +304,386 @@ public boolean unregisterDataListener(FreeboxDataListener dataListener) {
     }
 
-    private void setSambaPrinterStatus(Command command) throws FreeboxException {
-        if (command != null) {
-            if (command instanceof OnOffType || command instanceof OpenClosedType || command instanceof UpDownType) {
-                SambaConfig sambaConfiguration = new SambaConfig();
-
-                sambaConfiguration.setPrintShareEnabled(command.equals(OnOffType.ON) || command.equals(UpDownType.UP)
-                        || command.equals(OpenClosedType.OPEN));
-
-                fbClient.getNetShareManager().setSambaConfig(sambaConfiguration);
+    private void setSambaPrinterStatus(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType || command instanceof OpenClosedType || command instanceof UpDownType) {
+            try {
+                updateChannelSwitchState(SAMBAPRINTERSTATUS,
+                        apiManager.enableSambaPrintShare(command.equals(OnOffType.ON) || command.equals(UpDownType.UP)
+                                || command.equals(OpenClosedType.OPEN)));
+            } catch (FreeboxException e) {
+                logCommandException(e, channelUID, command);
                 fetchSambaConfig();
             }
+        } else {
+            logger.debug("Thing {}: invalid command {} from channel {}", getThing().getUID(), command,
+                    channelUID.getId());
         }
     }
 
-    private void setReboot(Command command) throws FreeboxException {
-        if (command != null) {
-            if (command.equals(OnOffType.ON) || command.equals(UpDownType.UP) || command.equals(OpenClosedType.OPEN)) {
-                fbClient.getSystemManager().Reboot();
+    private void reboot(ChannelUID channelUID, Command command) {
+        if (command.equals(OnOffType.ON) || command.equals(UpDownType.UP) || command.equals(OpenClosedType.OPEN)) {
+            try {
+                apiManager.reboot();
+            } catch (FreeboxException e) {
+                logCommandException(e, channelUID, command);
             }
+        } else {
+            logger.debug("Thing {}: invalid command {} from channel {}", getThing().getUID(), command,
+                    channelUID.getId());
         }
     }
 
+    private void updateChannelStringState(String channel, String state) {
+        updateState(new ChannelUID(getThing().getUID(), channel), new StringType(state));
+    }
+
+    private void updateChannelSwitchState(String channel, boolean state) {
+        updateState(new ChannelUID(getThing().getUID(), channel), state ? OnOffType.ON : OnOffType.OFF);
+    }
+
+    private void updateChannelDecimalState(String channel, int state) {
+        updateState(new ChannelUID(getThing().getUID(), channel), new DecimalType(state));
+    }
+
+    private void updateChannelDecimalState(String channel, long state) {
+        updateState(new ChannelUID(getThing().getUID(), channel), new DecimalType(state));
+    }
+
+    public void logException(FreeboxException e) {
```

Comment:
Hmm, I'm in doubt about this, rather add some context to the logging statement and put the on there own places.

Reply:
Done